precompiles/dapps-staking/DappsStaking.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+pragma solidity >=0.7.0;
+
+/// Interface to the precompiled contract on Shibuya/Shiden/Astar
+/// Predeployed at the address 0x0000000000000000000000000000000000005001
+interface DappsStaking {
+
+    // Storage getters
+
+    /// @dev Read current era.
+    /// @return The current era
+    function read_current_era() external view returns (uint256);
+
+    /// @dev Read unbonding period constant.
+    /// @return The unbonding period
+    function read_unbonding_period() external view returns (uint256);
+
+    /// @dev Read Total network reward for the given era
+    /// @return Total network reward for the given era
+    function read_era_reward(uint32 era) external view returns (uint128);
+
+    /// @dev Read Total staked amount for the given era
+    /// @return Total staked amount for the given era
+    function read_era_staked(uint32 era) external view returns (uint128);
+
+    /// @dev Read Staked amount for the staker
+    /// @return Staked amount for the staker
+    function read_staked_amount(address staker) external view returns (uint128);
```

Comment:
Probably, using H160 everywhere instead of contract id will make troubles in the future, but in other case we could just overload same method with bytes32 AccountId when needed.

Reply:
@akru do you mean to add following
  ` function read_staked_amount(uint256 staker) external view returns (uint128);`
We actually need it for `astarbase` since we need to verify 32bit public polkadot address